File: src/easy/#605. Can Place Flowers.js
// https://leetcode.com/problems/can-place-flowers/description/

function canPlaceFlowers(arr, n) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] !== 1 && arr[i + 1] !== 1 && arr[i - 1] !== 1) {
      n--;
      arr[i] = 1;
    }
  }

  return n <= 0;
}

File: src/easy/#605. Can Place Flowers.py
def canPlaceFlowers(arr: list[int], n: int) -> bool:
    arr.insert(0, 0)
    arr.append(0)

    i = 1

    while i < len(arr)-1:
        if arr[i] == 0 and arr[i-1] == 0 and arr[i+1] == 0:
            arr[i] = 1
            n = n - 1

        i = i + 1

    return n <= 0
